src/async_impl/client.rs

```diff
@@ -94,6 +94,7 @@ struct Config {
     tls_built_in_root_certs: bool,
     #[cfg(feature = "__tls")]
     tls: TlsBackend,
+    http1_only: bool,
```

Comment:
Since `http1_only` and `http2_only` are exclusive (they can't both be true), perhaps it'd be best if we made an internal enum to track that?

```rust
enum HttpVersionPref {
    Http1,
    Http2,
    All,
}
```